draft-ietf-quic-manageability.md

```diff
@@ -1023,6 +1023,38 @@ not arrive at their destination. Admitting a few packets allows the QUIC
 endpoint to determine that the path accepts QUIC. Sudden drops afterwards will
 result in slow and costly timeouts before abandoning the connection.
 
+## Guiding Path MTU
+
+Some networks are required to be able to carry 1500-byte packets but become
+inefficient at this size, because such packets are fragmented and
+reassembled across a segment where the effective MTU is smaller (e.g. due to
+IPSEC overhead). They do this even when the encapsulated packet is not a
+candidate for fragmentation by routers (it's IPv6, or has DF bit set). In such
+networks this leads to a need to influence a majority of senders to use smaller
+packets, so that the limited reassembly capacity is not exceeded.
+
+If the majority use TCP this can be managed with MSS Clamping, but QUIC
+requires a different approach. Section 14 of {{QUIC-TRANSPORT}} advises senders
+to probe larger sizes using Datagram Packetization Layer PMTU Discovery
+({{?DPLPMTUD=RFC8899}}) or Path Maximum Transmission Unit Discovery
+(PMTUD: {{?RFC1191}} and {{?RFC8201}}). This mechanism will encourage senders to
+approach the maximum size, driving fragmentation that they may not be aware of.
+
+Networks can seek to avoid this by identifying probable QUIC packets
+({{sec-identifying}}) and taking action whenever packets exceed the desired
+size. For DPLPMTUD a sufficient action is to drop the packet. If it is
+genuinely QUIC, the packet should have been a probe and so its loss should not
+cause impact to the flow of data. It serves only to guide the discovery of an
+optimal Path MTU.
+
+DPLPMTUD can optionally make use of ICMP Packet Too Big messages (PTB, defined
```

Comment:
When a device drops an IP packet, it ought to send an ICMP PTB message indicating the supported MTU, and quoting the discarded packet. However, endpoints do not necessarily receive these PTB Messages, and QUIC is robust to their loss.

Reply:
Agree. Are you suggesting the wording should be tightened, so that dropping the packet and not bothering with the PTB is called out as less-desirable?

Reply:
Now I see where this is heading, I think we should encourage PTB ... and perhaps say something like:

"({{sec-identifying}}) and taking action whenever packets exceed the desired size. For DPLPMTUD it is sufficient for the path to silently drop a packet larger than the supported size. (A QUIC probe packet is used to discover the PMTU. If lost, this does not impact the flow of QUIC data.) 

IPv4 routers generate an ICMP message when a packet is dropped because the link MTU was exceeded. {{RFC8504}} specifies how an IPv6 node generates an ICMPv6 Packet Too Big message (PTB) in this case. 

PMTUD relies upon an endpoint receiving such PTB messages {{RFC8201}},  whereas DPLPMTUD does not reply upon these messages, but still can optionally utilise these to improve performance {{Section 4.6 of DPLPMTUD}}. 

Since a network cannot know in advance which discovery method a QUIC endpoint is using, it ought to send a PTB message in addition to dropping any oversized packet.  A generated PTB message should be compliant with the validation requirements of {{section 14.2.1 of QUIC-TRANSPORT}}, otherwise it will be ignored by DPLPMTUD."

- choose: /ought to send a PTB message/needs to send/ - choose appropriate words!

Reply:
Details:
- Unwelcome detail: In some cases, a PMTUD implementation *might* accept any ICMP  message to update the IP cache, but that isn't a great choice, and text in recent RFCs seeks to discourage this.
- As far as I recall, the RFC series is very murky for IPv4. 
- For IPv6 the advice is clear, despite the current state of affairs reported in RFC8899, and that this alone can't really solve the problem. It says: "For IPv6, "a Packet Too Big MUST be sent by a router in response to a packet that it cannot forward because the packet is larger than the MTU of the outgoing link"{{RFC4443}}. These PTB messages "MUST NOT be filtered" {{RFC8504}}.
- We likely do not wish to cite that - because also it does not solve the problems of PTB message delivery, but we ought to strongly encourage PTB where it can work.

Reply:
Happy with the new wording. Regarding "ought to send" my inclination is to use SHOULD but I'm not sure if normative language is permitted here. Otherwise "should".

Reply:
The document is not presently not using RFC2119 language, so "should" would seem OK personally, from my side propose what seems best.

Reply:
Applied this in the latest commit.